.github/workflows/container-smoke-test.yml

```diff
@@ -13,12 +13,27 @@ jobs:
     steps:
       - name: Checkout repository
         uses: actions/checkout@v3
+        # Need fetch-depth 0 to fetch tags, see https://github.com/actions/checkout/issues/701
+        with:
+          fetch-depth: 0
 
-      - name: Build karapace/version.py
-        run: python version.py
+      - name: Install requirements
+        run: |
+          python -m venv venv
+          source ./venv/bin/activate
+          pip install .
 
-      - name: Build and start services
-        run: docker compose --file=container/compose.yml up --build --wait --detach
+      - name: Resolve Karapace version
+        run: |
+          source ./venv/bin/activate
+          KARAPACE_VERSION=$(python -c "from karapace import version; print(version.__version__)")
+          echo KARAPACE_VERSION=$KARAPACE_VERSION >> $GITHUB_ENV
+
+      - name: Build container
+        run: docker build --build-arg KARAPACE_VERSION=${{ env.KARAPACE_VERSION }} --file=container/Dockerfile .
+
+      - name: Run container
+        run: docker compose --file=container/compose.yml up --wait --detach
```

Comment:
issue: I think removing `--build` here is going to result in compose instead downloading `ghcr.io/aiven-open/karapace:develop`, rather than using a build with evaluated version of the code-base.

There are some controls for the behavior: https://docs.docker.com/reference/compose-file/build/#using-build-and-image. But I think we would also want to the above build command to rather invoke `docker compose build` than `docker build`.